### PR TITLE
Create fetch_plan in base

### PR DESCRIPTION
**Pull request by mari-linhares**
* Plans are no longer searchable :cry: . This is mainly due to the different behavior when "searching" for plans and searching for tensors.
* Now is possible to fetch a plan based on its id :smile: . This will get a plan from a worker's registry, create a copy of the plan and change the owner to whoever is fetching the plan.

This is the result of a [~~pair~~ triple programming section](https://github.com/OpenMined/PySyft/commit/66f5cc9688f5c74140772766420ba5d034d24c01) with @robert-wagner and @jlebensold.

### CODE REVIEW

File: syft/workers/base.py [hunk 1]
```diff
@@ -719,15 +719,37 @@ def request_remote_tensor_shape(self, pointer):
...
         return sy.hook.torch.Size(shape)
 
+    def fetch_plan(self, plan_id):
+        """Fetchs a copy of a the plan with the given `plan_id` from the worker registry.
+
+        Args:
+            plan_id: A string indicating the plan id.
+
+        Returns:
+            A plan if a plan with the given `plan_id` exists. Otherwhise returns None.
```
Comment (robert-wagner) [inline syft/workers/base.py]: Can you say that it returns none otherwise

File: syft/workers/base.py [hunk 2]
```diff
@@ -719,15 +719,37 @@ def request_remote_tensor_shape(self, pointer):
         shape = self.send_msg(MSGTYPE.GET_SHAPE, pointer, location=pointer.location)
         return sy.hook.torch.Size(shape)
 
+    def fetch_plan(self, plan_id):
```
Comment (robert-wagner) [inline syft/workers/base.py]: Can you add typing

File: syft/workers/base.py [hunk 3]
```diff
@@ -719,15 +719,37 @@ def request_remote_tensor_shape(self, pointer):
...
+        """Fetchs a copy of a the plan with the given `plan_id` from the worker registry.
+
+        Args:
+            plan_id: A string indicating the plan id.
+
+        Returns:
+            A plan if a plan with the given `plan_id` exists. Otherwhise returns None.
+        """
+        if plan_id in self._objects:
```
Comment (jlebensold) [inline syft/workers/base.py]: do we prefer indentation to [guard](https://blog.startifact.com/posts/none_05_more_on_guarding.html) statements?

Comment (mari-linhares) [reply syft/workers/base.py]: thanks for pointing out this blog! I think we can go with the guard statement. I'll fix it.

**Pull request merged**